bin/common.bash

```diff
@@ -146,11 +152,12 @@ yq_copy_changes() {
     source2=$2
     target=$3
 
-    keys=$(yq read "${source2}" --tojson --printMode pv '**' | \
-           sed -rn 's/\{"(.+)":.*\}/\1/p' | sed -r 's/\.\[.+\].*//' | sed -r 's/\\//g' | uniq)
+    keys=$(yq4 '.. | select(tag != "!!seq" and tag != "!!map") | path |
+           with(.[] | select(contains(".")); . = "\"" + . + "\"") |
+           join(".")' "${source2}" | \
+           sed -r 's/\.[0-9].*//' | sed -r 's/\\//g' | uniq)
```

Comment:
```suggestion
    keys=$(yq4 '.. | select(tag != "!!map") | path |
           with(.[]; . = ("\"" + .) + "\"" ) |
           join "."' "$source2" | \
           sed -r 's/\."[0-9]".*//' | uniq)
```
I think this can be simplified a bit. Not selecting on `!!seq` won't allow us to set `[]` to null an array which is a regression.
The same style can be used in `yq_copy_commons` and `yq_read_block`, maybe this should just be it's own function.